drivers/hwmon/pmbus/adm1266.c

```
@@ -548,6 +548,51 @@ static int adm1266_unlock_all_dev(struct adm1266_data *data)
 	return 0;
 }
 
+static int adm1266_lock_device(struct adm1266_data *data)
+{
+	struct i2c_client *client = data->client;
+	u8 passwd_cmd[ADM1266_PASSWD_CMD_LEN];
+	int reg_val;
+	int ret;
+
+	memset(passwd_cmd, 0, ADM1266_PASSWD_CMD_LEN);
+	passwd_cmd[ADM1266_PASSWD_CMD_LEN - 1] = ADM1266_LOCK_DEV;
+
+	ret = pmbus_block_write(client, ADM1266_FW_PASSWORD,
+				ADM1266_PASSWD_CMD_LEN,
+				passwd_cmd);
+	if (ret < 0)
+		return ret;
+
+	/* check if device is now locked */
+	reg_val = pmbus_read_byte_data(client, 0, ADM1266_STATUS_MFR);
+	if (reg_val < 0)
+		return reg_val;
+
+	if (!ADM1266_STATUS_PART_LOCKED(reg_val))
+		return -EBUSY;
+
+	return 0;
+}
+
+static int adm1266_lock_all_dev(struct adm1266_data *data)
+{
+	struct adm1266_data_ref *slave_ref;
+	int ret;
+
+	ret = adm1266_lock_device(data);
+	if (ret < 0)
+		return ret;
+
+	list_for_each_entry(slave_ref, &data->cascaded_devices_list, list) {
+		ret = adm1266_lock_device(slave_ref->data);
```

general question here?
what if one of the `adm1266_lock_device()` fails? how would we need to handle this?
in typical cases, you'd typically do a rollback;

but maybe for this case, it may make more sense to ignore the error, continue with locking other devices and print out a dev_warn() or dev_err() message;
maybe [in case of error]:
1. print an error message: could not lock deviceX
2. [optionally] print a final message like `dev_warn("Failed to lock 3 devices out of 10")`; in case there are more devices that could not be locked, this would indicate that the failure is intermittent